### PR TITLE
Add support for reconfig cluster membership operation (issue #234)

### PR DESCRIPTION
**Pull request by rgs1**
Signed-off-by: Raul Gutierrez S <rgs@itevenworks.net>

### CODE REVIEW

**Comment by rgs1** [comment]
this passes for me locally fwiw (given the tests aren't happy in travis)

File: kazoo/tests/test_client.py [hunk 1]
```diff
@@ -1147,3 +1148,57 @@ def test_unchroot(self):
...
+
+class TestReconfig(KazooTestCase):
+
+    def setUp(self):
+        KazooTestCase.setUp(self)
+        if TRAVIS_ZK_VERSION:
+            version = TRAVIS_ZK_VERSION
+        else:
+            version = self.client.server_version()
+        if not version or version < (3, 5):
```
Comment (harlowja) [inline kazoo/tests/test_client.py]: Should this kind of check be denoted in the reconfig docstring? (or should that function even do that check itself?)

Comment (rgs1) [inline kazoo/tests/test_client.py]: you mean, checking if the server supports it? you'd just UnimplementedError in that case

Comment (harlowja) [inline kazoo/tests/test_client.py]: Maybe that, or at least docstrings?

**Comment by Yasumoto** [comment]
This is really exciting- thanks for adding it!

**Pull request merged**